specification/frontdoor/resource-manager/Microsoft.Network/stable/2021-06-09/webapplicationfirewall.json

```diff
@@ -0,0 +1,1038 @@
+            "$ref": "#/definitions/ManagedRuleExclusion"
+          },
+          "description": "Describes the exclusions that are applied to this specific rule."
+        }
+      }
+    },
+    "ManagedRuleSetDefinitionList": {
+      "description": "List of managed rule set definitions available for use in a policy.",
+      "properties": {
+        "value": {
+          "description": "List of managed rule set definitions.",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleSetDefinition"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to retrieve next set of managed rule set definitions."
+        }
+      }
+    },
+    "ManagedRuleSetDefinition": {
+      "description": "Describes the a managed rule set definition.",
+      "properties": {
+        "properties": {
+          "description": "Properties for a managed rule set definition.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ManagedRuleSetDefinitionProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "ManagedRuleSetDefinitionProperties": {
+      "description": "Properties for a managed rule set definition.",
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Provisioning state of the managed rule set."
+        },
+        "ruleSetId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Id of the managed rule set."
+        },
+        "ruleSetType": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Type of the managed rule set."
+        },
+        "ruleSetVersion": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Version of the managed rule set type."
+        },
+        "features": {
+          "type": "object",
+          "readOnly": true,
+          "$ref": "#/definitions/Features",
+          "description": "Feature set applying to the Ruleset."
+        },
+        "supportedRuleActions": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ActionType"
+          },
+          "description": "Supported actions for the Rule."
+        },
+        "supportedRuleSetActions": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ActionType"
+          },
+          "description": "Supported actions for the entire RuleSet."
+        },
+        "supportedSkus": {
+          "description": "Map of objects for sku to related properties like state, etc.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/RuleSetDataBag"
+          }
```

Comment:
Should this just be an array? Not sure if the dynamic property names (SKUs) will cause issues for SDK generation.

Reply:
The PR auto-generates other PRs which trigger SDK generation. The .NET SDK actually doesn't look too bad. It created a IDictionary<string, RuleSetDataBag> data type. The potentially confusing part is that the string key is actually restricted to supported SKUs.

[https://github.com/AzureSDKAutomation/azure-sdk-for-net/pull/4378/files](https://github.com/AzureSDKAutomation/azure-sdk-for-net/pull/4378/files)

Reply:
oh cool, did not notice that. Thanks!
For the supported skus, is that not we want, to restrict them to the supported Skus only?

Reply:
Yes, but there's no inherit enforcement since it's generated as a string. It would be better if it was the SKU enum.

Reply:
Changed to use only the Sku values. Using an older format, but it seems to work.